### DISCUSSION TITLE
Cron workflow is not scheduled as expected

### DISCUSSION THREAD

**Discussion by zhangkyou** [uber/cadence#4383] [Q&A] [answered]
我的这个cadence任务调度周期设置的是每分钟都执行，但是看起来这15分钟都没有调度，我想查下这个为什么没调度，应该从哪里入手？
![image](https://user-images.githubusercontent.com/7250068/129854093-a711f825-4d05-4d91-8b56-47c5dea7dbf0.png)
workflow的options配置如下：
ExecutionStartToCloseTimeout: 120s
CronSchedule: "* * * * *"

activity的options配置如下：
ScheduleToStartTimeout: 20s
StartToCloseTimeout: 60s

从上面图中可以看到，不同run id的workflow的调度间隔，不是严格的每分钟一次，想问下这个问题该怎么着手查？

从cadence server中的日志看到了有一些这样的error
`{"level":"error","ts":"2021-08-17T23:25:13.432+0800","msg":"Persistent store operation failure","service":"cadence-matching","component":"matching-engine","wf-task-list-name":"cadence-worker-sf-a469b-4.docker.us01:60646bc2-e01c-49f1-8632-a3919c4d5884","wf-task-list-type":0,"store-operation":"update-task-list","error":"ServiceBusyError{Message: Persistence Max QPS Reached.}","logging-call-at":"taskReader.go:162","stacktrace":"github.com/uber/cadence/common/log/loggerimpl.(*loggerImpl).Error\n\t/Users/didi/go/src/go.uber.org/cadence-0.20.0/common/log/loggerimpl/logger.go:131\ngithub.com/uber/cadence/service/matching.(*taskReader).getTasksPump\n\t/Users/didi/go/src/go.uber.org/cadence-0.20.0/service/matching/taskReader.go:162"}`

**Comment by longquanzheng** [comment, marked as answer]
If previous execution hasn’t closed, it would delay the next run for cron workflow. This is to ensure the ID uniqueness guarantee and also the workflow timeout guarantee. 

If you wish to run the cron every minute, the workflow timeout must be less than or equal to a minute

**Comment by longquanzheng** [comment]
`Persistent store operation failure` on matching node means the service is maxing out the persistence ratelimiting. 
You can either add more nodes to matching service, or adjust the persistence [ratelimiting config](https://cadenceworkflow.io/docs/operation-guide/setup/#dynamic-configuration). 

See `MatchingPersistenceMaxQPS` or `MatchingPersistenceGlobalMaxQPS`. 

`MatchingPersistenceMaxQPS` is per node, `MatchingPersistenceGlobalMaxQPS` is global for all nodes of matching service. 

If `MatchingPersistenceGlobalMaxQPS` is configured, `MatchingPersistenceMaxQPS` will be ignored.
